src/base/searchengine.h

```diff
@@ -38,10 +40,119 @@
 class QProcess;
 class QTimer;
 
+template <class T, std::size_t N>
+class Version
```

Comment:
If you like, we can add another template parameter (number of mandatory printed digits) and change constructors and `operator QString()` accordantly.